lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -207,4 +212,58 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
     current_epoch = Misc.compute_epoch_at_slot(current_slot)
     store.justified_checkpoint.epoch + 1 == current_epoch
   end
+
+  @spec root_by_id(atom() | Types.root() | Types.slot()) ::
+          {:ok, {Types.root(), boolean(), boolean()}} | {:error, String.t()} | atom()
+  def root_by_id(:head) do
+    with {:ok, current_status} <- BeaconChain.get_current_status_message(),
+         {:ok, signed_block} <- BlockStore.get_block(current_status.head_root) do
+      # TODO compute is_optimistic_or_invalid
+      execution_optimistic = true
+      {:ok, {signed_block.message.state_root, execution_optimistic, false}}
+    end
+  end
+
+  def root_by_id(:genesis), do: :invalid_id
+
+  def root_by_id(:justified) do
+    with {:ok, justified_checkpoint} <- ForkChoice.get_justified_checkpoint(),
+         {:ok, signed_block} <- BlockStore.get_block(justified_checkpoint.root) do
+      # TODO compute is_optimistic_or_invalid
+      execution_optimistic = true
+      {:ok, signed_block.message.state_root, execution_optimistic, false}
+    end
+  end
+
+  def root_by_id(:finalized) do
+    with {:ok, finalized_checkpoint} <- ForkChoice.get_finalized_checkpoint(),
+         {:ok, signed_block} <- BlockStore.get_block(finalized_checkpoint.root) do
+      # TODO compute is_optimistic_or_invalid
+      execution_optimistic = true
+      {:ok, signed_block.message.state_root, execution_optimistic, true}
+    end
+  end
+
+  def root_by_id(:invalid_id), do: :invalid_id
+
+  def root_by_id(hex_root) when is_binary(hex_root) do
+    # TODO compute is_optimistic_or_invalid() and is_finalized()
+    execution_optimistic = true
+    finalized = false
+    {:ok, {hex_root, execution_optimistic, finalized}}
+  end
+
+  def root_by_id(slot) when is_integer(slot) do
+    with {:ok, state} <- StateStore.get_latest_state(),
+         {:ok, signed_block} <- BlockStore.get_block_by_slot(state.slot),
+         {:ok, store} <- get_forkchoice_store(state, signed_block.message),
+         current_slot = Store.get_current_slot(store),
+         true <- slot < current_slot,
```

Comment:
This is a bad way of checking. You should use a `check_x(...)` function that internally maps the `true`/`false` into `:ok`/`some_kind_of_error`. That way you don't just return `false` when this validation fails.